### DISCUSSION TITLE
Handling indirect jumps with the new lifter

### DISCUSSION THREAD

**Discussion by frabert** [lifting-bits/anvill#337] [Ideas]
Hey I decided to try using the Discussions feature to see if it works better than Issues for more conversational topics.

@2over12 How are indirect jumps going to be handled with regards to the new low-level lifter? Do we still generate jump intrinsics or do we just update the PC and let the overarching switch function do its thing?

**Comment by 2over12** [comment]
Currently, the jump intrinsic will still be generated. If we have the known targets in the Ghidra CFG then the parent function will have edges after the call to the target blocks. If not, there will be no successors. 

Conceptually, this is a bit weird because it means for a jump where we know the target we create a call to a basic block function that doesn't really return (because of the implicit semantics of the remill jump intrinsic) and also represent it's targets in the parent call after it returns

As to what we should do, I think we can't guarantee much about situation if we don't have the successors of a block so perhaps we should not emit the jump and just have it update PC. If we don't have targets that's an error. Indirect calls should be admissable though

**Reply by 2over12** [reply]
I actually think what I wrote above is inaccurate, currently we will just update PC and if the targets aren't in the Ghidra CFG then we wont have any successors in the parent function. This is probably close to what we want but perhaps we should be more explicit somehow about missing targets in Ghidra.
